### DISCUSSION TITLE
Templates - UX/UI changes

### DISCUSSION THREAD

**Discussion by drewcochrane** [guardian/grid#4044] [Ideas]
To look at the template flow in 'My recent upload' page and how we make sure a user can easily find templates as part of uploading their image and how we make sure the list of templates doesn't get too unusable.

We are aware the Guardian don't use templates at present but they're becoming increasingly useful for us, however, UX isn't quite right so we're taking a look

Potential solutions include:

- Moving the template drop down to the top of the workflow.
- When a user selects Newspaper another drop down appears to show the specific newspaper template.

As of the moment, the template dropdown is located underneath the image thumbnail.

<img width="1138" alt="My recent upload" src="https://user-images.githubusercontent.com/116095133/223482743-9f67d431-d671-49ea-a78d-1626dff3d16e.png">

Proposed, template moved to top of workflow, to improve the page hierarchy - 'Choose a template' label is not final (Figma design screen grab).

![Upload page](https://user-images.githubusercontent.com/116095133/223483373-8389a36a-75b1-4ea1-a382-e487407b9c95.png)

Our current template dropdown as it is today.
<img width="299" alt="Screenshot 2023-03-07 at 16 20 38" src="https://user-images.githubusercontent.com/116095133/223483782-6d652607-b803-44ce-975d-74482119bd6d.png">

What we would like to do is create an accessible dropdown with nested dropdown within it. Had a chat with the wider UX who got some feedback from the assistive technology team around how to make nested dropdown components accessible. This is a mid fidelity design proposal for the dropdown where a new second column opens up when you click on the 'chevron' icon for Newspaper from the default dropdown.

<img width="1380" alt="Sub nested dropdowns" src="https://user-images.githubusercontent.com/116095133/223485039-e3af825d-8ba7-4d50-a8d9-d46d9b8f0d18.png">

**Comment by paperboyo** [comment]
Yes, we don’t use them, but they were always a very interesting concept (and they exist in other similar apps). Major obstacle, at least in my opinion, was exactly what you are trying to address here: too many to choose from. Without being able to filter them for groups of users, what you propose (grouping them) seems to me most sensible!

I also very much agree and like their new placement: makes more sense.

Would we ever develop a custom dropdown like the one above (great to see search there!), we should think about reusing it in other places too (and it being extensible, so it works for other uses).

cc @pradasa

**Comment by pradasa** [comment]
This is really interesting. We don't use templates so I'm curious about the details about how you use them at BBC. 

Moving it to the top of the right column makes a lot of sense (I had initially missed it when it was under the thumbnail tbh). All the content about the image would now sit together in the same column. And adding a search bar + 2 levels of information will make it easier to navigate all the options given here are so many under newspaper. 
 
I agree with @paperboyo , it would be nice to think about how the dropdown could be reusable for other scenarios.

**Comment by drewcochrane** [comment]
@honorcb is the best person to give us the details around how we use Templates at BBC

**Comment by honorcb** [comment]
Is to help the uploaders, have a standard texts,  for additional information texts ,  add to correct collection.    The set we are creating that I think might make sense to the Guardian are when dealing with images to KILL from agencies,   Making sure we have consistent wording,  automatically  remove from collections,  set deny use leases, and set to NO RIGHTS  for the Killed images as one action ,by selecting the KILL template, rather remembering to do  5 or 6 actions

**Reply by honorcb** [reply]
Our News picture desk add the front pages for the newspapers everyday,  the system we are replacing ELVIS allowed them to add using "templates" to add the basic information .  BBC Images newspaper templates are adding,  which newspaper,  who owns the paper, set the Credit and Copyright,   add to the correct collection ,  and set the Rights , ( Composite)  and  allow use lease for 24hrs ,  Creating these in  BBC Images ,  added 15 templates to the list,  which is why we need to think about grouping, not selecting from a long dropdown list,

**Reply by paperboyo** [reply]
Thanks, Honor! Just regarding
> automatically remove from collections

this is cool! Would that be setting an empty Collection via template? (maybe you can share your template config file or parts of it with us to take a look?)

**Comment by paperboyo** [comment]
Am I missing something (very possible, coz we are not using them), but shouldn’t it be possible to batch apply a template? cc @drewcochrane @honorcb

**Reply by honorcb** [reply]
It would be great,  currently it a one by one allocation .   If you are using a template to add your batch of pictures ,  you can use the double end arrows to propagate the text and rights associate with the template you have selected for the first image in your batch,  but being able to correct batches of select images in grid view  would be really helpful
